### PR TITLE
offline-logs-improvements

### PR DESCRIPTION
**Pull request by rotem-codefresh**
- adjustments to older versions of mongodb
- added background option to index creation

### CODE REVIEW

File: lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js [hunk 1]
```diff
@@ -1,96 +1,278 @@
...
+    cutoffDateId,
+    collection
+  );
+
+  if (!minLog) {
+    console.info(
+      `No logs to archive in Collection ${collection} from the given date.`
+    );
+    return;
+  }
```
Comment (noam-codefresh) [inline lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js:81]: you can extract this bit to the parent function, and only have this code in a single place.

File: lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js [hunk 2]
```diff
@@ -1,96 +1,278 @@
...
+      `logs from '${collection}' were archived to '${targetCollection}'`
+    );
+    await sourceCollectionObj.deleteMany({
+      _id: { $lte: ObjectId(cutoffDateId) },
+    });
+  } else {
+    throw new Error(
+      "Cursor error. Archiving operation may not be completed.\
+    The old logs were not deleted from the source collection."
+    );
```
Comment (noam-codefresh) [inline lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js:108]: also - this bit can be extracted into some "checkCursorState" func

File: lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js (hunk 1 above)
Comment (rotem-codefresh) [reply lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js]: done

File: lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js (hunk 2 above)
Comment (rotem-codefresh) [reply lib/interface/cli/commands/offline-logs/offload-to-collection.cmd.js]: done

**Review by noam-codefresh — APPROVED** [review]

**Review by noam-codefresh — APPROVED** [review]

**Pull request merged**